server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -824,13 +837,19 @@ long readSnapshotIndexLatestBlob() throws IOException {
 
     private long listBlobsToGetLatestIndexId() throws IOException {
         Map<String, BlobMetaData> blobs = blobContainer().listBlobsByPrefix(INDEX_FILE_PREFIX);
-        long latest = RepositoryData.EMPTY_REPO_GEN;
         if (blobs.isEmpty()) {
             // no snapshot index blobs have been written yet
-            return latest;
+            return RepositoryData.EMPTY_REPO_GEN;
```

Comment:
I think you can pass the `blobs` directly to `latestGeneration()` which will take care of returning `EMPTY_REPO_GEN` and remove the block

```
if (blobs.isEmpty()) {
   // no snapshot index blobs have been written yet
   return RepositoryData.EMPTY_REPO_GEN;
}
```